docs/apps/hyperqueue.md

```diff
@@ -0,0 +1,241 @@
+# HyperQueue
+
+!!!warning "Autoallocation"
+    The autoallocation feature available in HQ is still under development and buggy, don't use it
+    as it's very likely that the job queue will be filled with idling workers which just waste
+    resources.
+
+HyperQueue (HQ) is a tool for efficient sub-node task scheduling. Instead of submitting each one of
+your computational tasks using `sbatch` or `srun` you can instead allocate a large resource block
+and then use HyperQueue to submit your tasks. This will be much less stressful for the batch queue
+system and is therefore the recommended way to run your high-throughput computing use cases.
+
+## License
+
+Free to use and open source under [MIT License](https://github.com/It4innovations/hyperqueue/blob/main/LICENSE)
+
+## Available
+
+* Puhti: 0.4.0, 0.9.0
+* Mahti: 0.11-dev
+
+## Usage
+
+``
+module load hyperqueue
+``
+
+HyperQueue works on a worker-server-client basis. Here, the server manages connections, workers are
+started on compute nodes and execute commands which the client submitted to the server. This
+resembles a Slurm within a Slurm, but you have to start the server and workers yourself.
+
+### Starting the server
+
+!!! info "Note"
+    The following instructions apply for Slurm job scripts where only full nodes
+    are allocated. [A full example](#full-example) (for Mahti) can be found at the bottom.
+
+Specify where the HyperQueue server should be placed. All `hq` commands respect this variable
```

Comment:
Maybe we could write that the server is holding its state on the filesystem. Not sure if this is clear for a newbie from the text, at least not early enough in the text.

Also, if we are advertising multinode runs as well, we could mention that it needs to be on `lustre`? 🤔 Not sure if this one is necessary, since we probably do not want to suggest even that they could put it on `/dev/shm` for example.

Reply:
Sure, modified the text a tad to better reflect that the server is on the file system (notably, on lustre).